CHANGELOG.md

```diff
@@ -1,6 +1,53 @@
+* If you use any programming/scripting logic to list and select a minor version of Kubernetes before creating clusters with the `ListKubernetesVersions` API, note that starting from Kubernetes v1.27, the API returns `SupportPlan` as `[KubernetesOfficial, AKSLongTermSupport]`. Please ensure you update any logic to exclude `AKSLongTermSupport` versions to avoid any breaks and choose `KubernetesOfficial` support plan versions.  Otherwise, if LTS is indeed your path forward please first opt-into the Premium tier and the `AKSLongTermSupport` support plan versions from the `ListKubernetesVersions` API before creating clusters.  Refer [long term support](https://learn.microsoft.com//azure/aks/long-term-support) for more information.
+* AKS patch version 1.29.4 is now available.
+
+### Release Notes
+
+* Features:
+  * With this release, Azure Linux 2.0 is a supported OS for AKS Long Term Support (LTS) with v1.27. Learn more about [Azure Linux and LTS](https://aka.ms/aks-azurelinux-lts).
+  * You can now get insight into the progress of any ongoing operation, such as create, upgrade, and scale, using any preview API version after `2024-01-02-preview` using the Get/List operations call. Refer to [Long running operations on an Azure Kubernetes Service (AKS) cluster](https://learn.microsoft.com/azure/aks/manage-abort-operations?tabs=azure-cli) for more information.
+    
+* Behavioral Changes:
+  * Manually added Labels, Taints, and Annotations on nodes will no longer be copied to nodes during surged upgrade. To ensure any Label or Taint is present in new nodes please use the [Labels and/or Taints](https://learn.microsoft.com/azure/aks/use-labels) functionality provided by AKS. 
+  * The Istio-based service mesh add-on now skips validation of its compatibility with cluster version unless mesh upgrade or cluster upgrade is attempted.
+  * The Istio-based service mesh add-on components can now tolerate running on the system node pools with the `CriticalAddonsOnly` taint.
+  
+* Bug Fixes:
+  * Fixed a bug that incorrectly calculated number of free IPs in a subnet when upgrading an agent pool using Azure CNI with Dynamic IP allocation.
+  * Fixed a bug to allow correct IP address calculation for subnets with Private Link Service.
+  * Fixed a bug where the ordering of the system environment variables injected into pods could change.
+  * Fixed a bug in clusters that use [Node Autoprovisioning](https://learn.microsoft.com/azure/aks/node-autoprovision) for stateful workloads deployments that use availability zones.
+  * Fixed a bug in clusters that use [Node Autoprovisioning](https://learn.microsoft.com/azure/aks/node-autoprovision) and managed identity to authenticate Azure Container Registry.
+  * Fixed an issue for clusters older than v1.20.X where Cluster Autoscaler cannot be started or reconciled.
+  * Fixed an issue where the `agentPoolProfiles` property was getting wiped on Azure CNI Overlay clusters using pod identity.
```

Comment:
let's improve the language here. "property was getting deleted". Also which property? agentPoolProfile is a full profile of properties.
is this actually pod identity or workload identity?